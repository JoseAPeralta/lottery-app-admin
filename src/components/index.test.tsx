import { render } from '@testing-library/react';
import { axe } from 'vitest-axe';
import { LotteryList } from './index.ts';

describe('Component LotteryBoard', () => {
  const LotteryListProps = [
    {
      id: '63b3820ce784d28ac9d57c9b',
      drawDate: new Date('2022-12-30T05:00:00Z'),
      folio: 6,
      letters: 'BDDD',
      prizes: {
        first: '2013',
        second: '39',
        third: '39',
      },
      serie: 2,
      type: 'Zodiaco',
    },
    {
      id: '63b381bee784d28ac9d57c9a',
      drawDate: new Date('2022-12-28T05:00:00Z'),
      folio: 6,
      letters: 'BDAA',
      prizes: {
        first: '3189',
        second: '1633',
        third: '5915',
      },
      serie: 20,
      type: 'Intermedio',
    },
    {
      id: '63b38190e784d28ac9d57c99',
      drawDate: new Date('2022-12-24T05:00:00Z'),
      folio: 11,
      letters: 'AABA',
      prizes: {
        first: '6098',
        second: '1339',
        third: '9971',
      },
      serie: 15,
      type: 'Dominical',
    },
    {
      id: '63b380dfe784d28ac9d57c97',
      drawDate: new Date('2022-12-21T05:00:00Z'),
      folio: 5,
      letters: 'DDBA',
      prizes: {
        first: '4637',
        second: '1238',
        third: '5439',
      },
      serie: 8,
      type: 'Intermedio',
    },
    {
      id: '63b38159e784d28ac9d57c98',
      drawDate: new Date('2022-12-18T05:00:00Z'),
      folio: 1,
      letters: 'DCDD',
      prizes: {
        first: '0667',
        second: '6284',
        third: '2642',
      },
      serie: 19,
      type: 'Dominical',
    },
  ];

  it('Should match to snapshot', () => {
    const result = render(<LotteryList lotteryDraws={LotteryListProps} />);
    expect(result).toMatchSnapshot();
  });

  it('Should accesible LotteryBoard', async () => {
    const { container } = render(<LotteryList lotteryDraws={LotteryListProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
