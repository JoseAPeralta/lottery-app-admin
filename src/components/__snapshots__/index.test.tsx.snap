// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component LotteryBoard > Should match to snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Hi!
      </h1>
      <section>
        <section>
          <div>
            Zodiaco
          </div>
          <div>
            30
          </div>
          <div>
            2013
          </div>
          <div>
            39
          </div>
          <div>
            39
          </div>
          <div>
            BDDD
          </div>
          <div>
            2
          </div>
          <div>
            6
          </div>
        </section>
        <section>
          <div>
            Intermedio
          </div>
          <div>
            28
          </div>
          <div>
            3189
          </div>
          <div>
            1633
          </div>
          <div>
            5915
          </div>
          <div>
            BDAA
          </div>
          <div>
            20
          </div>
          <div>
            6
          </div>
        </section>
        <section>
          <div>
            Dominical
          </div>
          <div>
            24
          </div>
          <div>
            6098
          </div>
          <div>
            1339
          </div>
          <div>
            9971
          </div>
          <div>
            AABA
          </div>
          <div>
            15
          </div>
          <div>
            11
          </div>
        </section>
        <section>
          <div>
            Intermedio
          </div>
          <div>
            21
          </div>
          <div>
            4637
          </div>
          <div>
            1238
          </div>
          <div>
            5439
          </div>
          <div>
            DDBA
          </div>
          <div>
            8
          </div>
          <div>
            5
          </div>
        </section>
        <section>
          <div>
            Dominical
          </div>
          <div>
            18
          </div>
          <div>
            0667
          </div>
          <div>
            6284
          </div>
          <div>
            2642
          </div>
          <div>
            DCDD
          </div>
          <div>
            19
          </div>
          <div>
            1
          </div>
        </section>
      </section>
      u
    </div>
  </body>,
  "container": <div>
    <h1>
      Hi!
    </h1>
    <section>
      <section>
        <div>
          Zodiaco
        </div>
        <div>
          30
        </div>
        <div>
          2013
        </div>
        <div>
          39
        </div>
        <div>
          39
        </div>
        <div>
          BDDD
        </div>
        <div>
          2
        </div>
        <div>
          6
        </div>
      </section>
      <section>
        <div>
          Intermedio
        </div>
        <div>
          28
        </div>
        <div>
          3189
        </div>
        <div>
          1633
        </div>
        <div>
          5915
        </div>
        <div>
          BDAA
        </div>
        <div>
          20
        </div>
        <div>
          6
        </div>
      </section>
      <section>
        <div>
          Dominical
        </div>
        <div>
          24
        </div>
        <div>
          6098
        </div>
        <div>
          1339
        </div>
        <div>
          9971
        </div>
        <div>
          AABA
        </div>
        <div>
          15
        </div>
        <div>
          11
        </div>
      </section>
      <section>
        <div>
          Intermedio
        </div>
        <div>
          21
        </div>
        <div>
          4637
        </div>
        <div>
          1238
        </div>
        <div>
          5439
        </div>
        <div>
          DDBA
        </div>
        <div>
          8
        </div>
        <div>
          5
        </div>
      </section>
      <section>
        <div>
          Dominical
        </div>
        <div>
          18
        </div>
        <div>
          0667
        </div>
        <div>
          6284
        </div>
        <div>
          2642
        </div>
        <div>
          DCDD
        </div>
        <div>
          19
        </div>
        <div>
          1
        </div>
      </section>
    </section>
    u
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
